File: src/main/java/com/dh/ondot/schedule/api/response/PlaceSearchResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.schedule.application.dto.PlaceSearchResult;

import java.util.List;

public record PlaceSearchResponse(
        String title,
        String roadAddress,
        Double longitude,
        Double latitude
) {
    public static PlaceSearchResponse from(PlaceSearchResult result) {
        return new PlaceSearchResponse(
                result.getTitle(),
                result.getRoadAddress(),
                result.getLongitude(),
                result.getLatitude()
        );
    }

    public static List<PlaceSearchResponse> fromList(List<PlaceSearchResult> results) {
        return results.stream()
                .map(PlaceSearchResponse::from)
                .toList();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/api/response/PlaceSearchResponse.kt
package com.dh.ondot.schedule.api.response

import com.dh.ondot.schedule.application.dto.PlaceSearchResult

data class PlaceSearchResponse(
    val title: String?,
    val roadAddress: String?,
    val longitude: Double?,
    val latitude: Double?,
) {
    companion object {
        @JvmStatic
        fun from(result: PlaceSearchResult): PlaceSearchResponse {
            return PlaceSearchResponse(
                title = result.title,
                roadAddress = result.roadAddress,
                longitude = result.longitude,
                latitude = result.latitude,
            )
        }

        @JvmStatic
        fun fromList(results: List<PlaceSearchResult>): List<PlaceSearchResponse> {
            return results.map { from(it) }
        }
    }
}
